### DISCUSSION TITLE
[BO] Improve UX for List of changed files

### DISCUSSION THREAD

**Discussion by mparvazi** [PrestaShop/PrestaShop#41068] [Ideas]
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md)
- [X] I have already [searched in existing features request](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3AImprovement%2CFeature%2CDeveloper+Feature+) and found no previous suggestion of this feature.

### Is your feature request related to a problem?

Current report display of "List of changed files" doesn't show useful information.

![Screenshot 2022-02-10 at 15-26-16 Information • develop](https://user-images.githubusercontent.com/85633460/153404168-b8ef7595-509f-45aa-b71e-ce64d57f04ae.png)

### Describe the solution you'd like

Some useful information should be added:

![Screenshot 2022-02-10 at 15-21-13 Information • develop](https://user-images.githubusercontent.com/85633460/153403599-50be9865-c014-49e3-8ac6-f29d08679ca8.png)

**Last modified:** The user can see when the file has been changed. For example shop had unexpected behavior in 2 past days, so user want to see which files in Core have been modified in this time period.

### Do you plan to work on this feature?

- [ ] I'm willing to contribute a [formal specification](https://github.com/PrestaShop/prestashop-specs).
- [ ] I'm willing to provide any wireframes or design assets required for this feature.
- [X] I'm willing to submit a Pull Request that implements this feature.
- [X] I'm willing to help verify that the implemented feature works as intended and produces no unintended side effects.

**Comment by marwachelly** [comment]
Hello,

Thank you for your suggestion. The Product Team will take it into consideration for future developments. 
Please be aware that there is no guarantee that this feature will be developed anytime soon. If this is important to you, we strongly suggest you consider [hiring a specialist](https://www.prestashop.com/en/experts) to help you create it. 

If you are willing to implement this feature into PrestaShop as a [Pull Request](https://devdocs.prestashop.com/1.7/contribute/contribute-pull-requests/), the teams will be happy to help you by ensuring that it is appropriate and coherent with the rest of the project. If that is the case, please let us know!

Thanks,

**Comment by Hlavtox** [comment]
@mparvazi What would be cool would be an "extract" button. You click it and it would copy the changed files into a zip for download. A) Backup, B) Easier transfer of mods to new version.

**Comment by mparvazi** [comment]
https://user-images.githubusercontent.com/85633460/153401985-eaca10d6-d0b9-480d-82ca-87cee6296f75.mp4

**Comment by PululuK** [comment]
Hi @mparvazi 
Good ideia ... it might be interesting to see the difference between the original files ?

**Comment by mparvazi** [comment]
> Hi @mparvazi Good ideia ... it might be interesting to see the difference between the original files ?

Hi @PululuK 

I thought about it, but how we can show original file for current patch?

**A. GitHub:**
For example PS:1.7.8.3, we have 1.7.8.x repo for original files, but this files may update since 1.7.8.3 release.

**B. Release patch archives:**
```
<?php
$za = new ZipArchive(); 
$za->open('https://github.com/PrestaShop/PrestaShop/releases/download/1.7.8.3/prestashop_1.7.8.3.zip'); 
for( $i = 0; $i < $za->numFiles; $i++ ){ 
    $stat = $za->statIndex( $i ); 
    print_r( basename( $stat['name'] ) . PHP_EOL ); 
}
?>
```
But it will increase load time.

**Comment by marsaldev** [comment]
Very interesting idea!

> I thought about it, but how we can show original file for current patch?
How about adding a button and start the check? With a 'warning' or something similar to warn the user that it will take time?

@mparvazi 
I'm available to help with the PR if you want 😉
